lib/remap-lang/src/value/kind.rs

```diff
@@ -29,6 +29,12 @@ impl Kind {
     pub fn scalar(self) -> Self {
         self & !(Kind::Array | Kind::Map)
     }
+
+    /// Returns `true` if the [`value::Kind`] is a scalar and `false` if it's
+    /// map or array.
+    pub fn is_scalar(self) -> bool {
+        !matches!(self, Kind::Array | Kind::Map)
+    }
```

Comment:
Could we add a unit test to make sure this is working as expected?

(we don't have any in this module yet, but we really should, that's something I should have done earlier).

What we want to make sure happens is:

* If `Kind` only contains scalar kinds, return `true` (e.g. `Boolean`, or `Boolean | Bytes`, etc)
* Otherwise return `false` (e.g. `Map`, or `Boolean | Map`, etc)

Reply:
Also, could we reuse `scalar()` here rather? Something like `self == self.scalar()`?